### PR TITLE
Api pagination

### PR DESCRIPTION
**Pull request by liviuba**
https://app.clickup.com/t/8695j4dpq

### CODE REVIEW

File: api/api/models/api.py [hunk 1]
```diff
@@ -18,3 +19,8 @@ class TokenData(BIABaseModel):
 class AuthResult(BIABaseModel):
     access_token: str
     token_type: Literal["bearer"] = "bearer"
+
+
+class Pagination(BIABaseModel):
+    start_uuid: Optional[UUID] = None
```
Comment (sherwoodf) [inline api/api/models/api.py:25]: suggest renaming to 'start_from_uuid'

**Review by sherwoodf — CHANGES_REQUESTED** [review]
Having issues testing locally due to having wiped my workspace after the filename issues. I'm also getting pylance issues with two types that are passed in as variables - will try to figure this out tomorrow, but neither of those have anything to do with your changes here.

**Comment by sherwoodf** [comment]
Currently getting 2 failing tests locally:
- test_get_studies
- test_search_image_multiple_results
But these are also failing on main...

File: api/api/tests/test_pagination.py [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+    rsp = api_client.get(
+        f"study/{existing_study['uuid']}/experimental_imaging_dataset",
+        params={"page_size": 3},
+    )
+    assert rsp.status_code == 200
+    first_page = rsp.json()
+
+    rsp = api_client.get(
+        f"study/{existing_study['uuid']}/experimental_imaging_dataset",
+        params={"page_size": 3, "start_uuid": first_page[-1]["uuid"]},
```
[the remark this answers is not in the document]
Comment (liviuba) [reply api/api/tests/test_pagination.py:106]: Not sure I understand the suggestion, in general page_size sets the maximum size of the response, the client doesn't need to know ahead of time if they would get `page_size` results or less.

[This](https://github.com/BioImage-Archive/bia-integrator/pull/219/files#diff-1a8bbf7b53e35f14f2959bfd88c7e3dc603b9b8f1dac086ad2129b92ac59d881R43) covers cases where the client has a page size larger than the whole result set, and [this](https://github.com/BioImage-Archive/bia-integrator/pull/219/files#diff-1a8bbf7b53e35f14f2959bfd88c7e3dc603b9b8f1dac086ad2129b92ac59d881R94) covers the typical case where the client gets multiple pages (not knowing the total), until a page is smaller than page_size.

Happy to add the extra tests, but I don't know which scenario isn't covered

**Review by sherwoodf — APPROVED** [review]
LGTM

**Pull request merged**